src/datasets/io/parquet.py

```diff
@@ -101,6 +115,13 @@ def _write(self, file_obj: BinaryIO, batch_size: int, **parquet_writer_kwargs) -
         written = 0
         _ = parquet_writer_kwargs.pop("path_or_buf", None)
         schema = pa.schema(self.dataset.features.type)
+
+        schema = schema.with_metadata(
+            ArrowWriter._build_metadata(
+                info=DatasetInfo(features=self.dataset.features)
+            )
+        )
+
```

Comment:
this way you already get a schema with metadata :)
```suggestion
        schema = self.dataset.features.arrow_schema
```

Reply:
ah, wow, that's much simpler :)